File: dataset/CoT - Copy/GEMINI_gemini-2.5-pro/Task59.cpp
#include <iostream>
#include <string>
#include <cctype>

class Task59 {
public:
    static std::string accum(const std::string& s) {
        std::string result = "";
        for (size_t i = 0; i < s.length(); ++i) {
            // Append the first character, capitalized
            result += toupper(static_cast<unsigned char>(s[i]));
            // Append the rest of the characters, in lowercase, i times
            for (size_t j = 0; j < i; ++j) {
                result += tolower(static_cast<unsigned char>(s[i]));
            }
            // Append a hyphen if it's not the last part
            if (i < s.length() - 1) {
                result += "-";
            }
        }
        return result;
    }
};

int main() {
    std::string test_cases[] = {"abcd", "RqaEzty", "cwAt", "Z", ""};
    for (const auto& test_case : test_cases) {
        std::cout << "\"" << test_case << "\" -> \"" << Task59::accum(test_case) << "\"" << std::endl;
    }
    return 0;
}

File: dataset/ZeroShot/OPENAI_gpt-5-2025-08-07/Task59.java
public class Task59 {
    public static String accum(String s) {
        if (s == null) return "";
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (i > 0) sb.append('-');
            sb.append(Character.toUpperCase(c));
            char lower = Character.toLowerCase(c);
            for (int k = 0; k < i; k++) sb.append(lower);
        }
        return sb.toString();
    }

    public static void main(String[] args) {
        String[] tests = new String[] {"abcd", "RqaEzty", "cwAt", "", "Z"};
        for (String t : tests) {
            System.out.println(accum(t));
        }
    }
}